gqcp/include/Processing/Properties/DysonOrbital.hpp

```diff
@@ -0,0 +1,155 @@
+// This file is part of GQCG-GQCP.
+//
+// Copyright (C) 2017-2020  the GQCG developers
+//
+// GQCG-GQCP is free software: you can redistribute it and/or modify
+// it under the terms of the GNU Lesser General Public License as published by
+// the Free Software Foundation, either version 3 of the License, or
+// (at your option) any later version.
+//
+// GQCG-GQCP is distributed in the hope that it will be useful,
+// but WITHOUT ANY WARRANTY; without even the implied warranty of
+// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+// GNU Lesser General Public License for more details.
+//
+// You should have received a copy of the GNU Lesser General Public License
+// along with GQCG-GQCP.  If not, see <http://www.gnu.org/licenses/>.
+
+#pragma once
+
+#include "ONVBasis/SpinResolvedONVBasis.hpp"
+#include "QCModel/CI/LinearExpansion.hpp"
+
+
+namespace GQCP {
+
+
+/**
+ *  Dyson orbital.
+ */
+template <typename Scalar>
+class DysonOrbital {
+private:
+    VectorX<Scalar> dyson_coefficients;
+
+
+public:
+    /*
+     *  MARK: Constructors
+     */
+
+
+    /**
+     *  The default constructor.
+     */
+    DysonOrbital() = default;
+
+    DysonOrbital(const size_t K) :
+        dyson_coefficients(VectorX<Scalar>::Zero(K)) {}
+
+    DysonOrbital(const VectorX<Scalar> dyson_amplitudes) :
+        dyson_coefficients {dyson_amplitudes} {}
+
+    /*
+     *  MARK: Named constructors
+     */
+
+    /**
+     *  Calculate the coefficients of a Dyson orbital, expressed as the overlap between two wave functions expressed in the same spinor basis.
+     * 
+     *  @param linear_expansion1        A wave function in a spin-unresolved ONV basis.
+     *  @param linear_expansion2        A wave function in a spin-unresolved ONV basis containing one fewer electron and the same number of orbitals that is expressed in the same basis
+     *
+     *  @return A Dyson orbital containing Dyson amplitudes.
+     */
+    static DysonOrbital<Scalar> calculateDysonOrbital(const LinearExpansion<SpinResolvedONVBasis>& linear_expansion1, const LinearExpansion<SpinResolvedONVBasis>& linear_expansion2) {
```

Comment:
```suggestion
    static DysonOrbital<Scalar> Overlap(const LinearExpansion<SpinResolvedONVBasis>& linear_expansion1, const LinearExpansion<SpinResolvedONVBasis>& linear_expansion2) {
```

Possibly `FromOverlap`?

Also, the documentation seems incomplete: are you calculating `<N-1|a_p|N>` or something else?